tabbycat/draw/generator/powerpair.py

```diff
@@ -469,9 +469,9 @@ def _pullup(self, brackets, indices):
 
         # Tuples: (teams_list, side, number_needed)
         # List by highest bracket first.
-        pullups_needed_for = list()
+        pullups_needed_for: list = list()
```

Comment:
Hey, thanks for doing this. Can you give us a rundown on why type hints like this one are useful? I know it's legal, but my understanding was the type hints were mainly to be used when it's not obvious from context (mainly with function parameters and return values, which I definitely agree we could do with more of, especially with functions/methods called from other files), and for static type checkers (which should be able to figure out this one, since `list()` can't be anything other than a `list`).

Reply:
The type here could be inferred – I like making types as explicit as possible, but that's just personal preference. My feeling is that having only partial typing means that not everything has a type (that's not `any`). While I agree that here you could infer that `pullups_needed_for` has type list, type annotations (using `typing.List`) would be useful to work out what the list is a list of.